server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -891,6 +892,16 @@ public void deleteStaleSegments(int lastNMetadataFilesToKeep) throws IOException
             lastSuccessfulFetchOfPinnedTimestamps
         );
 
+        if (metadataFilesEligibleToDelete.isEmpty()) {
+            logger.debug("No metadata files are eligible to be deleted based on lastNMetadataFilesToKeep and age");
+            return;
+        }
+
+        // If pinned timestamps are enabled, make sure to not delete last metadata file.
+        if (RemoteStoreSettings.isPinnedTimestampsEnabled()) {
```

Comment:
should we have a check of `lastNMetadataFilesToKeep  > 0` as well to delete it once the index is deleted ? This might not be an ideal check , but this will have some parity with `RemoteFsTranslog` as well .

Reply:
This is not required here as latest metadata file is already removed if `lastNMetadataFilesToKeep  > 0`